src/cli/tree.rs

```diff
@@ -0,0 +1,396 @@
+                if direct_deps.contains(&pkg.name) {
+                    console::style(pkg.name.clone()).fg(Color::Green).bold()
+                } else {
+                    console::style(pkg.name.clone())
+                },
+                pkg.version
+            );
+
+            print_inverted_leaf(pkg, String::from(""), inverted_dep_map, direct_deps);
+        }
+    }
+
+    Ok(())
+}
+
+/// Recursively print inverted dependency tree leaf nodes
+fn print_inverted_leaf(
+    pkg: &InvertedPkg,
+    prefix: String,
+    inverted_dep_map: &HashMap<String, InvertedPkg>,
+    direct_deps: &Vec<String>,
+) {
+    let needed_count = pkg.needed_by.len();
+    for (index, needed_name) in pkg.needed_by.iter().enumerate() {
+        let last = index == needed_count - 1;
+        let symbol = if last {
+            UTF8_SYMBOLS.ell
+        } else {
+            UTF8_SYMBOLS.tee
+        };
+
+        if let Some(needed_pkg) = inverted_dep_map.get(needed_name) {
+            println!(
+                "{}{} {} v{}",
+                prefix,
+                symbol,
+                if direct_deps.contains(&needed_pkg.name) {
+                    console::style(needed_pkg.name.clone())
+                        .fg(Color::Green)
+                        .bold()
+                } else {
+                    console::style(needed_pkg.name.clone())
+                },
+                needed_pkg.version,
+            );
+
+            let new_prefix = if index == needed_count - 1 {
+                format!("{}{} ", prefix, UTF8_SYMBOLS.empty)
+            } else {
+                format!("{}{} ", prefix, UTF8_SYMBOLS.down)
+            };
+
+            print_inverted_leaf(needed_pkg, new_prefix, inverted_dep_map, direct_deps)
+        }
+    }
+}
+
+/// Filter and print a top down dependency tree
+fn print_dependency_tree(
+    dep_map: &HashMap<String, Pkg>,
+    direct_deps: &[String],
+    regex: &Option<String>,
+) -> Result<(), miette::Error> {
+    let mut direct_deps = direct_deps.to_owned();
+
+    if let Some(regex) = regex {
+        let regex = regex::Regex::new(regex).map_err(|_| miette::miette!("Invalid regex"))?;
+        direct_deps.retain(|p| regex.is_match(p));
+
+        if direct_deps.is_empty() {
+            Err(miette::miette!(
+                "No top level dependencies matched the given regular expression"
+            ))?;
+        }
+    }
+
+    let mut visited_pkgs = Vec::new();
+    let direct_dep_count = direct_deps.len();
+
+    for (index, pkg_name) in direct_deps.iter().enumerate() {
+        visited_pkgs.push(pkg_name.to_owned());
+
+        let last = index == direct_dep_count - 1;
+        let symbol = if last {
+            UTF8_SYMBOLS.ell
+        } else {
+            UTF8_SYMBOLS.tee
+        };
+        if let Some(pkg) = dep_map.get(pkg_name) {
+            println!(
+                "{} {} v{}",
+                symbol,
+                console::style(pkg.name.clone()).fg(Color::Green).bold(),
```

Comment:
_Brain dump_

Not mandatory for a first release but do you have ideas to make it more clear which package is a pypi package and which is a conda package in the tree?
Some ideas that come to my mind.
- Simply add `pypi::` or `conda::` to the name, maybe a bit verbose?
- Using emoji's, possibly not supported everywhere.
- Giving a different color, maybe not clear at first glance but could be added to the `--help` 

I guess these ideas should be synced with the `list` command.

Reply:
What about coloring the version numbers?

Reply:
I quickly gave colored version numbers a shot.

<img width="378" alt="image" src="https://github.com/prefix-dev/pixi/assets/1296209/e2526853-73a7-4759-8f5b-14df71c5d33e">

Reply:
Oh, that helped me see that I was displaying the parent's version number in leafs 
<img width="477" alt="image" src="https://github.com/prefix-dev/pixi/assets/1296209/1eec02c4-139e-4299-bb0d-91b9ceef6148">

I cleaned that up and pushed the package name & version styling into a single function, rather than rewriting it for each tree function. It also helped me realize I only needed one `Package` struct, I could just start with an empty `needed_by` vector.